x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/terms/values_range_input.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { useDebounce } from 'react-use';
+import { i18n } from '@kbn/i18n';
+import { EuiRange } from '@elastic/eui';
+
+type PropType<C> = C extends React.ComponentType<infer P> ? P : unknown;
+// Add ticks to EuiRange component props
+const FixedEuiRange = (EuiRange as unknown) as React.ComponentType<
+  PropType<typeof EuiRange> & {
+    ticks?: Array<{
+      label: string;
+      value: number;
+    }>;
+  }
+>;
+
+export const ValuesRangeInput = ({
+  value,
+  onChange,
+}: {
+  value: number;
+  onChange: (value: number) => void;
+}) => {
+  const MIN_NUMBER_OF_VALUES = 1;
+  const MAX_NUMBER_OF_VALUES = 100;
```

Comment:
I'm okay with setting it 100 based on our "best practices" case, but we could also set it based on the `MAX_BUCKETS` ui setting. What do you think?

Reply:
I like the idea of having it in the ui settings, but right now the set value for `metrics:max_buckets` is 2000. I checked in Visualize that we indeed allow terms of that many values, but the rendering is so slow (in my computer around a minute, if it doesn't stall completely) that it's completely unusable. And as it's easier to just use EuiRange to accidentally slide the value up than manually type it as we do in Visualize, I'd rather stick to maximum of 100 for lens. We could also add it as a separate setting in advanced settings if you think it's worth it.

Reply:
I would go forward with the PR in its current state as it's an improvement over the behavior in the old version. Long term we can rethink this UI (maybe a slider without text input and a free input as an "advanced mode"?) - but IMHO that's out of scope for this PR.